mlir/lib/Dialect/SparseTensor/Transforms/SparseIterationToScf.cpp

```diff
@@ -49,6 +50,144 @@ convertIteratorType(IteratorType itTp, SmallVectorImpl<Type> &fields) {
   return success();
 }
 
+static ValueRange
```

Comment:
It feels like this code can still be broken up into smaller utility methods to improve readabilty (probably later okay), but it is really cool to have all the loop logic generation in one file now!!

Reply:
SG, will do it later!